### ISSUE TITLE
Result without JaguarResponse

### ISSUE THREAD

**Issue by jaumard** [Jaguar-dart/jaguar_http#7]
In retrofit it's possible to have this kind of signature (for people who don't care about the raw response) : 

```
//Normal one with jaguarReponse
@Get("/users/:id")
Future<JaguarResponse<User>> getUserById(@Param() String id);

//Short one without jaguarReponse
@Get("/users/:id")
Future<User> getUserById(@Param() String id);
```

Is it currently possible ? Are you open to such possibility if I make a PR ?

**Comment by tejainece** [comment]
Yes! That would be very cool.

**Comment by jaumard** [comment]
Ok thanks ! I'll try to make it work first ^^ because for now I can't make it work as I have only Flutter installed the cli is not working (as it want dart command line which I don't have :/)

**Comment by tejainece** [comment]
The package is a bit outdated. I am updating it. I will let you know when I am done. So you can fork it.

**Comment by jaumard** [comment]
Any news on this one ? I'll soon need it for my project so I'll dig into it :) thx !

**Comment by tejainece** [comment]
Hi,

Sorry! Been busy with Jaguar videos, website and ORM. Gonna work on it this week. Will keep you posted.

**Comment by jaumard** [comment]
I saw that :) that's good! no problem at all ;)

**Comment by jaumard** [comment]
Can I help you with this @tejainece ? I need to start making http calls in my app and can't use it as it is for now as it doesn't work ^^ 
So if I can do some stuff to help you with this let me know ;) happy to help

**Comment by tejainece** [comment]
Would you like to try `jaguar_resty` or `jaguar_client`?. `jaguar_resty` is almost declarative and has automatic serialization using `jaguar_serializer`.

**Comment by jaumard** [comment]
@tejainece jaguar_resty look nice :) but I guess if I use it it will be to create another package to mimic retrofit style and use resty under the hood ^^, I'm coming from Java/Android world and I'm use to have just annotations to describe my API calls. That's why I'm offering you to work on this package ^^ as I'll anyway do something similar :) I prefer working on this one instead of doing one on my own and have duplicate package once this one will be up to date.

**Comment by tejainece** [comment]
Me too. I like retrofit style clients. Its just I am caught up with ORM and Jaguar examples.

Yes. The plan is to build Jaguar_http on top of jaguar_resty. jaguar_resty is almost like jaguar_query for jaguar_orm.

I will put some effort into jaguar_http tomorrow and day after. 

Meantime pull requests are welcome! Maybe I will do base layout later today.

**Comment by jaumard** [comment]
Totally understand that you're busy, no problem :) glad you work on ORM too, I'll dig a little into resty and see if I can start doing stuff for a PR ;) 
Thanks !

**Comment by tejainece** [comment]
Hi,

ORM is coming up great. Gonna start work on `jaguar_http` today.

**Comment by tejainece** [comment]
Changed `jaguar_http` to `declarative_client`. It resides in https://github.com/Jaguar-dart/client/tree/master/declarative_client.

It is still is in early stage. Do share your suggestions and opinions.

**Comment by jaumard** [comment]
Thanks, I'll take a look :) 
I'll say for Header and Query to add a `defaultValue`, to be able to provide something like `@Header('Accept', value: 'whatever')` instead of providing the value at runtime if it doesn't change at all :) 

I guess the generation will work like the ORM with build_runner ? Because I don't think it's the case currently with jaguar_http. 

I guess the repo itself is just interfaces, most of the job will be generation on this one, still obscur to me how this generation works ^^

**Comment by tejainece** [comment]
Yes. It will use `build_runner`. Will finish the interfaces over the next couple of days. Been busy with `sqljocky` code.

**Comment by jaumard** [comment]
@tejainece I see you've rename it back to jaguar_http (https://github.com/Jaguar-dart/client/tree/master/jaguar_http) is it ready to use ? What's the status ?

**Comment by tejainece** [comment]
Yes. I think to keep familiat jaguar_http is better.

It is usable but not published to the pub yet. You can clone local copies and try it. There are couple of examples.

I will continue to work on it through out this week.

PRs are welcome.

**Comment by jaumard** [comment]
jaguar_http is better I agree, thanks I'll test it asap :) 
Will do PR too if needed for sure
